hal/armv7m/mpu.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Phoenix-RTOS loader
+ *
+ * MPU routines
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <lib/lib.h>
+#include <lib/log.h>
+#include <lib/errno.h>
+#include <hal/hal.h>
+#include <syspage.h>
+#include "mpu.h"
+
+enum {
+	mpu_fError = 0x01,
+	mpu_fConfigured = 0x04,
+	mpu_fValid = 0x08,
+};
```

Comment:
This isn't our style too (will need clang-off off, sadly)

Reply:
Done.